### DISCUSSION TITLE
Query parameter validation

### DISCUSSION THREAD

**Discussion by judgej** [livewire/livewire#5634] [3. Feature ideas]
This is something that has bothered me from the start, and I have n ot found a way around it so far. I guess it equally applies to non-query parameter data, which *can* be manipulated in the browser, even if it is signed en-route.

Basically, any public property for a component is exposed to the front end. Some are updated by binding to items. Some are updated by binding them to query parameters. The problem is, Livewire sets those public parameters from data coming in before any kind of validation can be applied.

So, if I have a public property, say `public int $myThingId;` and a non-integer is passed in, then an exception is thrown when Livewire attempts to set the integer property to that string. That's just a standard PHP exception, and not something that can be handled from the component (so far as I know).

So, what can be done to allow bad data at this stage (e.g. `...?my-thing-id=abc&...`) to be caught and handled in a better way. I may just want to display an error page. I may want to clear that data item or defaut it. But a "type mismatch" error is not so useful.

**Comment by judgej** [comment]
For query parameters, a validation element for `$queryString` could be a good place to add rules, and perhaps to pass a closure to decide what to do if validation fails.

For non-query data it's a bit more complicated on how to handle it, but validation rules could still apply.

At the moment the only real way to handle it is to make every public property type-less, and to validate them (and cast them) in `mount()` and `hydrate()`.

Of course, I could be missing something altogether!

**Comment by milosb793** [comment]
+1 For this!

**Comment by dmitry-ivanov** [comment]
+1

It would be really nice to have validations for query parameters.
_(or to have them going through the same `updating` / `updated` hooks, as usual props)..._

**Comment by makroxyz** [comment]
+1

I have date (from - to) filter (query string)  and I need to validate that interval must not be greater than 30 days...

**Comment by firer777** [comment]
u can use Livewire\Validate attribute for validate like, https://wirebox.app/b/4oe65 and can validate again more every request by method boot in Livewire component

**Reply by jacob1237** [reply]
It doesn't work if you manually change the query string in the browser.

**Comment by makroxyz** [comment]
@sinhnv1991 Thank you very much... need to add keep: true!!
